models/inventory/tests/testInventory.spec.ts

```diff
@@ -90,6 +119,55 @@ test('create stock movement, material receipt', async (t) => {
   t.equal(await fyo.db.getStockQuantity(itemMap.Ink.name), quantity);
 });
 
+test('Batch Enabled : create stock movement, material receipt', async (t) => {
+  const { rate } = itemMap.Pen;
+  const quantity = 2;
+  const batchNumber = batchNumberMap.batchNumberOne.name;
+  const amount = rate * quantity;
+  const stockMovement = await getStockMovement(
+    MovementType.MaterialReceipt,
+    new Date('2022-11-03T09:57:04.528'),
+    [
+      {
+        item: itemMap.Pen.name,
+        to: locationMap.LocationOne,
+        quantity,
+        batchNumber,
+        rate,
+      },
+    ],
+    fyo
+  );
+
+  await (await stockMovement.sync()).submit();
+  t.ok(stockMovement.name?.startsWith('SMOV-'));
+  t.equal(stockMovement.amount?.float, amount);
+  t.equal(stockMovement.items?.[0].amount?.float, amount);
+
+  const name = stockMovement.name!;
+
+  const sles = await getSLEs(name, ModelNameEnum.StockMovement, fyo);
+  t.equal(sles.length, 1);
+
+  const sle = sles[0];
+  t.notEqual(new Date(sle.date).toString(), 'Invalid Date');
+  t.equal(parseInt(sle.name), 2);
+  t.equal(sle.item, itemMap.Pen.name);
+  t.equal(parseFloat(sle.rate), rate);
+  t.equal(sle.quantity, quantity);
+  t.equal(sle.location, locationMap.LocationOne);
+  t.equal(
```

Comment:
This is the only check of importance in this test.

Other than this, you should've also tested whether there's stock of any other batch.